packages/matchbox/src/components/Button/styles.js

```diff
@@ -0,0 +1,136 @@
+import { tokens } from '@sparkpost/design-tokens';
+
+export const base = (props) => `
+    display: inline-block;
+    border-radius: ${tokens.borderRadius_100};
+    user-select: none;
+    font-weight: ${tokens.fontWeight_medium};
+    text-decoration: none;
+    white-space: nowrap;
+    vertical-align: middle;
+    transition-property: background, color, border, outline;
+    transition-duration: .15s;
+    border: 1px solid transparent;
+  `;
+
+export const size = (props) => {
+  switch (props.visualSize) {
+    case 'large':
+      return `
+        line-height: calc(3.5rem - 2px);
+        font-size: ${tokens.fontSize_300};
+        padding: 0 ${tokens.spacing_500};
+      `;
+    case 'small':
+      return `
+        line-height: calc(${tokens.spacing_600} - 2px);
+        font-size: ${tokens.fontSize_200};
+        padding: 0 ${tokens.spacing_400};
+      `;
+    case 'default':
+    default:
+      return `
+        line-height: calc(2.5rem - 2px);
+        font-size: ${tokens.fontSize_200};
+        padding: 0 ${tokens.spacing_400};
+      `;
+  }
+};
+
+export const colorVariant = (props) => {
+  let color;
+  let hoverColor;
+
+  switch (props.buttonColor) {
+    case 'blue':
+      color = tokens.color_blue_700;
+      hoverColor = tokens.color_blue_800;
+      break;
+
+    case 'red':
+      color = tokens.color_red_700;
+      hoverColor = tokens.color_red_800;
+      break;
+
+    case 'gray':
+    default:
+      color = tokens.color_gray_900;
+      hoverColor = tokens.color_gray_1000;
+      break;
+  }
+
+  switch (props.visualWeight) {
+    case 'strong':
+      return `
+        &, &:visited {
+          background: ${color};
+          color: ${tokens.color_white};
+    
+          :not([disabled]):hover {
```

Comment:
This does increase the CSS specificity considerably, making overrides more difficult. Could the hover disabling be applied to the disabled styles only rather than sort of disabling the disabled hover state here?

Reply:
Good idea 👍